test/sparse/sparse.jl

```diff
@@ -945,886 +1004,951 @@ function test_getindex_algs{Tv,Ti}(A::SparseMatrixCSC{Tv,Ti}, I::AbstractVector,
+    @test Array(triu!(copy(A), 2)) == triu(AF,2)
+    @test Array(tril!(copy(A), 2)) == tril(AF,2)
+    @test_throws BoundsError tril(A,6)
+    @test_throws BoundsError tril(A,-6)
+    @test_throws BoundsError triu(A,6)
+    @test_throws BoundsError triu(A,-6)
+    @test_throws ArgumentError tril!(sparse([1,2,3], [1,2,3], [1,2,3], 3, 4), 4)
+    @test_throws ArgumentError tril!(sparse([1,2,3], [1,2,3], [1,2,3], 3, 4), -3)
+    @test_throws ArgumentError triu!(sparse([1,2,3], [1,2,3], [1,2,3], 3, 4), 4)
+    @test_throws ArgumentError triu!(sparse([1,2,3], [1,2,3], [1,2,3], 3, 4), -3)
+
+    # fkeep trim option
+    @test isequal(length(tril!(sparse([1,2,3], [1,2,3], [1,2,3], 3, 4), -1).rowval), 0)
+end
+
+@testset "norm" begin
+    A = sparse(Int[],Int[],Float64[],0,0)
+    @test norm(A) == zero(eltype(A))
+    A = sparse([1.0])
+    @test norm(A) == 1.0
+    @test_throws ArgumentError norm(sprand(5,5,0.2),3)
+    @test_throws ArgumentError norm(sprand(5,5,0.2),2)
+end
+
+@testset "ishermitian/issymmetric" begin
+    let
+        # real matrices
+        A = speye(5,5)
+        @test ishermitian(A) == true
+        @test issymmetric(A) == true
+        A[1,3] = 1.0
+        @test ishermitian(A) == false
+        @test issymmetric(A) == false
+        A[3,1] = 1.0
+        @test ishermitian(A) == true
+        @test issymmetric(A) == true
+
+        # complex matrices
+        A = speye(5,5) + im*speye(5,5)
+        @test ishermitian(A) == false
+        @test issymmetric(A) == true
+        A[1,4] = 1.0 + im
+        @test ishermitian(A) == false
+        @test issymmetric(A) == false
+
+        A = speye(Complex128, 5,5)
+        A[3,2] = 1.0 + im
+        @test ishermitian(A) == false
+        @test issymmetric(A) == false
+        A[2,3] = 1.0 - im
+        @test ishermitian(A) == true
+        @test issymmetric(A) == false
+
+        A = sparse(zeros(5,5))
+        @test ishermitian(A) == true
+        @test issymmetric(A) == true
+
+        # explicit zeros
+        A = speye(Complex128, 5,5)
+        A[3,1] = 2
+        A.nzval[2] = 0.0
+        @test ishermitian(A) == true
+        @test issymmetric(A) == true
+
+        # 15504
+        m = n = 5
+        colptr = [1, 5, 9, 13, 13, 17]
+        rowval = [1, 2, 3, 5, 1, 2, 3, 5, 1, 2, 3, 5, 1, 2, 3, 5]
+        nzval = [0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 1.0, 0.0, 1.0, 1.0, 1.0, 0.0, 1.0, 1.0, 1.0]
+        A = SparseMatrixCSC(m, n, colptr, rowval, nzval)
+        @test issymmetric(A) == true
+        A.nzval[end - 3]  = 2.0
+        @test issymmetric(A) == false
+
+        # 16521
+        @test issymmetric(sparse([0 0; 1 0])) == false
+        @test issymmetric(sparse([0 1; 0 0])) == false
+        @test issymmetric(sparse([0 0; 1 1])) == false
+        @test issymmetric(sparse([1 0; 1 0])) == false
+        @test issymmetric(sparse([0 1; 1 0])) == true
+        @test issymmetric(sparse([1 1; 1 0])) == true
+    end
+end
+
+@testset "Equality" begin
```

Comment:
Presumably the author included the "==" to identify the particular form of equality being tested, there being being multiple? Preserve that information?